apps/minifront/src/components/shared/non-native-fee-warning.tsx

```diff
@@ -0,0 +1,49 @@
+import { BalancesResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { Metadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { getAssetIdFromValueView } from '@penumbra-zone/getters/value-view';
+import { getAssetId } from '@penumbra-zone/getters/metadata';
+import { useStakingTokenMetadata } from '../../state/shared';
+import { ReactNode } from 'react';
+
+const hasStakingToken = (
+  balancesResponses: BalancesResponse[] = [],
+  stakingAssetMetadata?: Metadata,
+): boolean => {
+  return balancesResponses.some(asset =>
+    getAssetIdFromValueView
+      .optional()(asset.balanceView)
+      ?.equals(getAssetId.optional()(stakingAssetMetadata)),
```

Comment:
Weirdly, this is how Prettier formatted these lines — with a `?` at the beginning